### ISSUE TITLE
cloudbreak does not run recipe

### ISSUE THREAD

**Issue by jomach** [hortonworks/cloudbreak#2589]
We are trying to deploy cloudbreak on a vpc. And the yum commands can be only run over a proxy. 

We used the configuration  explained at: http://sequenceiq.com/cloudbreak-docs/latest/recipes/#sample-recipe-for-yum-proxy-setting

and it does not work. When I log on the ec2 the yum.conf ist not modified. 
the minion logs say : 
`017-06-20 08:34:21,829 [salt.loaded.int.module.cmdmod              ][ERROR   ][6730] stderr: http://public-repo-1.hortonworks.com/ambari/centos7/2.x/updates/2.4.0.1/repodata/repomd.xml: [Errno 12] Timeout on http://public-repo-1.hortonworks.com/ambari/centos7/2.x/updates/2.4.0.1/repodata/repomd.xml: (28, 'Connection timed out after 30001 milliseconds')
Trying other mirror.
s

 One of the configured repositories failed (Ambari 2.4.0.1-1),
 and yum doesn't have enough cached data to continue. At this point the only
 safe thing yum can do is fail. There are a few ways to work "fix" this:

     1. Contact the upstream for the repository and get them to fix the problem.

     2. Reconfigure the baseurl/etc. for the repository, to point to a working
        upstream. This is most often useful if you are using a newer
        distribution release than is supported by the repository (and the
        packages for the previous distribution release still work).

     3. Disable the repository, so yum won't use it by default. Yum will then
        just ignore the repository until you permanently enable it again or use
        --enablerepo for temporary usage:

            yum-config-manager --disable AMBARI.2.4.0.1-1

     4. Configure the failing repository to be skipped, if it is unavailable.
        Note that yum will try to contact the repo. when it runs most commands,
        so will have to try and fail each time (and thus. yum will be be much
        slower). If it is a very temporary problem though, this is often a nice
        compromise:

            yum-config-manager --save --setopt=AMBARI.2.4.0.1-1.skip_if_unavailable=true

failure: repodata/repomd.xml from AMBARI.2.4.0.1-1: [Errno 256] No more mirrors to try.`

Can someone help ? 
thx

**Comment by mhmxs** [comment]
Hi @jomach ,

The problem might be there is no internet on the node. Without internet access Cloudbreak couldn't provision Ambari cluster. You have to reconfigure your vpc to enable outgoing, NAT Gateway should be a good solution.

**Comment by mhmxs** [comment]
Hi @jomach 

Could you please check the result of the recipe execution? Does `/etc/yum.conf` look ok? Doeas proxy reachable from node?

**Comment by mhmxs** [comment]
Hi @jomach ,

Sorry it't too hot here to thinking :) Could you please attach Cloudbreak log around cluster provision? What version of Cloudbreak do you use?

**Comment by mhmxs** [comment]
You can find recipes under `/opt/scripts/pre/` is your recipe there?

**Comment by jomach** [comment]
We are using 1.6. 
[Cloudbreak_log.txt](https://github.com/sequenceiq/cloudbreak/files/1089352/Cloudbreak_log.txt)
I will check the recipes tomorrow

**Comment by jomach** [comment]
the folder : /opt/scripts/pre/  where does it is ? on cloudbreak dockers ? or in die ec2 instances ?

**Comment by mhmxs** [comment]
Cloudbreak must to upload recipes to cluster nodes into `/opt/scripts/pre/`. Are you sure you selected your recipe for all the hostgroups in the cluster, because i don't see any log message about the recipe upload. Upload recipes and execute them is two separated process, so one of them has to fail.

**Comment by jomach** [comment]
Yes, 100% sure I select that.

**Comment by mhmxs** [comment]
Is it possible to upgrade Cloudbreak to the latest?

**Comment by jomach** [comment]
it is the lastest... at least from the images avaible on aws.

**Comment by mhmxs** [comment]
The latest one is 1.16.1. You can use `cbd update` command to upgrade. Before you do this, would be nice to create a backup of the database before update, located under `/var/lib/cloudbreak` directory.

**Comment by jomach** [comment]
my version is already 1.16.1. I will upload a screen shot
![version](https://user-images.githubusercontent.com/4804546/27481279-2671b69a-581c-11e7-9a43-29c5e22b892c.JPG)

**Comment by doktoric** [comment]
Hi @jomach
The problem with the proxy configuration that we do not configure the proxy on the deployed clusters.
If you want to configure it on your cluster then you should do it before the salt bootstrap.
You should define this variable:
```
-Dcb.customData
```
you can define it in you profile with: 
```
export CB_JAVA_OPTS="-Dcb.customData='touch /tmp/testdata'"
```

in this case the variable which you define here is a script which will run before the bootstrap process so it will configure the proxy on the cluster.

Br,
R

**Comment by jomach** [comment]
@doktoric:  So you mean that the example provided http://sequenceiq.com/cloudbreak-docs/latest/recipes/#sample-recipe-for-yum-proxy-setting is wrong ?  but thanks for the info I will try that.

**Comment by doktoric** [comment]
@jomach the problem that the example only valid if your tool install is come after the recipe ran. In your case the ambari install itself is before the recipe run so you have to solve somehow the proxy config before that. I will create Jira ticket to implement the custom proxy configuration on the cluster.

**Comment by jomach** [comment]
@doktoric  we should remove the example with the proxy then. It is error prone imho. Or it should just be executed before the yum install ambari-server (that was what I was expecting of one PRE Recipe)

Regards

**Comment by jomach** [comment]
@doktoric so If I do something like this :export CB_JAVA_OPTS='-Dhttp.proxyHost=webproxy.co... -Dhttp.proxyPort=8080 -Dhttps.proxyHost=webproxy... -Dhttps.proxyPort=8080 -Dhttp.nonProxyHosts=172.17.0.1|*.service.consul|*.node.dc1.consul -Dcb.customData="sudo echo proxy=http://webproxy...:8080 >> /etc/yum.conf"' it does not work and it does not start properly, it misses all translations etc... 

One question about this Variable.  In your example how do you copy /tmp/testdata to the ec2 ? 
I could have something like Dcb.customData="sh configure_proxy.sh" but where do I put this file ?

**Comment by doktoric** [comment]
Did this command:
```
 -Dcb.customData="sudo echo proxy=http://webproxy...:8080 >> /etc/yum.conf
```

modified  /etc/yum.conf and then the install commands are not working or the file also not modified.

Regarding my example that is just a touch on the machine so if the file does not exist then it creates a new file. 
Br,
R

**Comment by jomach** [comment]
@doktoric if I add this property on Profile the cloud break does not start properly and I get no translations on the admin interface...

**Comment by doktoric** [comment]
@jomach will check on the weekend and will back to you
sorry for the delayed answer.
Br,
R

**Comment by jomach** [comment]
@doktoric  Any news ?

**Comment by doktoric** [comment]
@jomach sorry for the late answer. I find the bug which is an escaping problem. Trying to asap and will let you know

**Comment by jomach** [comment]
page not found on the pr. On which release will it be ?

**Comment by doktoric** [comment]
@jomach I dont know when the next release comes out but that will be the 2.1.0.

**Comment by jomach** [comment]
thx

**Comment by bwestpha** [comment]
Any news on that? 
My machines will not run and can not be set up if behind a proxy ... yum will never finish the updates and therefore cloudbreak wont work at all.

**Comment by doktoric** [comment]
@bwestpha it was fixed but you should wait for 2.1.0.
If you use 1.16.x please let me know

**Comment by bwestpha** [comment]
@doktoric currently I tried to evaluate cloudbreak (version 1.16) on openstack. Since I didn’t get it run I stopped here. 

It would be nice to know how long it will take to publish v 2.1.0 ?! Is there a roadmap or a rough  date for the release? Or is there a other solution I can may try ? (Would be nice to have a clue how cloudbreak CAN work. But for now I could not get any cluster working)

Thanks 
Ben
